src/material/stepper/testing/step-harness.ts

```diff
@@ -0,0 +1,116 @@
+/**
+ * @license
+ * Copyright Google LLC All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.io/license
+ */
+
+import {ComponentHarness, HarnessLoader, HarnessPredicate} from '@angular/cdk/testing';
+import {StepHarnessFilters} from './step-harness-filters';
+
+/** Harness for interacting with a standard Angular Material step in tests. */
+export class MatStepHarness extends ComponentHarness {
+  /** The selector for the host element of a `MatStep` instance. */
+  static hostSelector = '.mat-step-header';
+
+  /**
+   * Gets a `HarnessPredicate` that can be used to search for a `MatStepHarness` that meets
+   * certain criteria.
+   * @param options Options for filtering which steps are considered a match.
+   * @return a `HarnessPredicate` configured with the given options.
+   */
+  static with(options: StepHarnessFilters = {}): HarnessPredicate<MatStepHarness> {
+    return new HarnessPredicate(MatStepHarness, options)
+        .addOption('label', options.label,
+            (harness, label) => HarnessPredicate.stringMatches(harness.getLabel(), label))
+        .addOption('selected', options.selected,
+            async (harness, selected) => (await harness.isSelected()) === selected)
+        .addOption('completed', options.completed,
+            async (harness, completed) => (await harness.isCompleted()) === completed)
+        .addOption('invalid', options.invalid,
+            async (harness, invalid) => (await harness.hasErrors()) === invalid);
+  }
+
+  /** Gets the label of the step. */
+  async getLabel(): Promise<string> {
+    return (await this.locatorFor('.mat-step-text-label')()).text();
+  }
+
+  /** Gets the `aria-label` of the step. */
+  async getAriaLabel(): Promise<string|null> {
+    return (await this.host()).getAttribute('aria-label');
+  }
+
+  /** Gets the value of the `aria-labelledby` attribute. */
+  async getAriaLabelledby(): Promise<string|null> {
+    return (await this.host()).getAttribute('aria-labelledby');
+  }
+
+  /** Whether the step is selected. */
+  async isSelected(): Promise<boolean> {
+    const host = await this.host();
+    return (await host.getAttribute('aria-selected')) === 'true';
+  }
+
+  /** Whether the step has been filled out. */
+  async isCompleted(): Promise<boolean> {
+    const state = await this._getIconState();
+    return state === 'done' || (state === 'edit' && !(await this.isSelected()));
+  }
+
+  /**
+   * Whether the data inside the step has any errors. Note that this will only match if showing
+   * errors has been enabled through the `showErrors` option in `STEPPER_GLOBAL_OPTIONS`.
+   */
+  async hasErrors(): Promise<boolean> {
```

Comment:
I'm iffy about including this one, because it doesn't mean that there are invalid form controls inside the step, but rather that there are invalid controls, the error state matcher determined that errors should be shown and `showError` was set to `true` via the `STEPPER_GLOBAL_OPTIONS` injection token which is false by default. I'm open to removing it since I can see how it could confuse someone that doesn't know about the stepper internals.

Reply:
I'm okay with this as is, though maybe expand the comment a bit to make it super explicit